### ISSUE TITLE
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module:

### ISSUE THREAD

**Issue by Romick2005** [Surnet/swagger-jsdoc#249]
After installing v.7.0.0-rc.2 cannot start the project, because of invalid ES module import used.
![image](https://user-images.githubusercontent.com/17779660/108480921-3685bc00-72a0-11eb-9c44-2f5583a7dc76.png)

**Comment by kalinchernev** [comment]
@Romick2005 which version of node.js are you using?
Have you checked https://github.com/Surnet/swagger-jsdoc#nodejs-version-requirements-commonjs-and-esm?
Did you check the suggestion made by the error handler in the message you shared?

**Comment by Romick2005** [comment]
node v12.18.1
npm 6.14.5

**Comment by Romick2005** [comment]
index.js in v7 uses import instead of require.
v7:
![image](https://user-images.githubusercontent.com/17779660/108483097-fbd15300-72a2-11eb-8487-1fe67536050f.png)
v6:
![image](https://user-images.githubusercontent.com/17779660/108483181-14da0400-72a3-11eb-94d6-c327bb27f9c3.png)

**Comment by kalinchernev** [comment]
It's the library that becomes a module in 7.x. It'd be useful for consumers who need ESM and I don't think it's your case.
In your case, you are trying to use the library with require, which tells node.js to resolve modules as CommonJS.

You can use 6.x which has module.exports, yes.

**Comment by kalinchernev** [comment]
It's unfortunately quite messy the cjs vs esm and it has been in the last couple of years.
See this, I hope it'll be useful as a general knowledge https://nodejs.org/api/esm.html
A solution from the library would be to use a transpilation step before publishing so to provide both esm and cjs.
I don't have the time for another rewriting at the moment though. It's frustrating for me as a library maintainer and for the consumers, but again: you don't need to bump to v7 if you don't need ESM

**Comment by Romick2005** [comment]
I thinl it is bettter to go with require and import map in package.json for version 7 to support both CommonJS and ES Modules.
Like this:
```
"exports": {
    "require": "./index.js",
    "import": "./esm/wrapper.js"
}

```
where wrapper.js looks like:
```
import cjsModule from '../index.js';
export const foo = cjsModule.foo;
```

Note that it’s easy to write an ESM wrapper for CJS libraries, but it’s not possible to write a CJS wrapper for ESM libraries.

**Comment by kalinchernev** [comment]
If you can make an example, please open a pull request, I welcome ideas

**Comment by kalinchernev** [comment]
I notice an increased activity (likes :) on comments here, but i don't know whether this means there is a better way or people come and see a solution of their problem.

Any suggestions how to handle this better than the 7.x rewrite or shall I close?

**Comment by KoteKotlyarov** [comment]
@kalinchernev I just downgrade `swagger-jsdoc` to 6.x. Otherwise it **looks like** I need to add `{ "type": "module" }` into my `package.json` and use `import/export` everywhere in my project.

The first solution is purrfectly fine with me.

**Comment by Divlo** [comment]
@kalinchernev I don't think we should use ESM modules as it's really new in the Node.js ecosystem.
If we use ESM, we should also provide compatibility with CommonJS.

Also there's another issue, the `v7` version is released under the latest tag, so when I run `npm install swagger-jsdoc`, it automatically install `v7` but instead it should install `v6`, since it's a 'rc' (release candidate).

**Comment by kalinchernev** [comment]
Hi @Divlo thank you for your feedback. Unfortunately, I feel that the weeks I spent making ESM version of the library are causing more confusion than benefit although my attempts to provide the 2 options natively with documentation and tests, suffering from the drawbacks of being on the leading edge.

For the moment, I keep this issue open, reply to issues, etc. but I already feel tired with the idea of yet another rewrite or a bundler setup which will stray the project off the main goal of being native node.js

As a "workaround", please use version 6 for CJS and 7 for ESM

**Comment by BinaryBen** [comment]
Just spent hours trying to figure out what was wrong because the rc is released via the latest tag. Please fix before releasing v7 and please put v6 on the latest tag.

**Comment by Divlo** [comment]
I mostly agree that we should gradually migrate CommonJS modules to ESM.
I really think it is the right think to do, I really wish, CommonJS didn't existed at all or that the whole ecosystem already migrated.

Excuse me for this

**Comment by a-dams154** [comment]
> @kalinchernev I just downgrade `swagger-jsdoc` to 6.x. Otherwise it **looks like** I need to add `{ "type": "module" }` into my `package.json` and use `import/export` everywhere in my project.
> 
> The first solution is purrfectly fine with me.

its not working for me

**Comment by a-dams154** [comment]
Any solution for this..?

**Comment by a-dams154** [comment]
> > @kalinchernev I just downgrade `swagger-jsdoc` to 6.x. Otherwise it **looks like** I need to add `{ "type": "module" }` into my `package.json` and use `import/export` everywhere in my project.
> > The first solution is purrfectly fine with me.
> 
> its not working for me

now its working for me downgrading swagger js-doc to 6.0.1

**Comment by premacck** [comment]
I was facing this issue, downgrading to 6.0.7 worked for me

**Comment by lachieh** [comment]
> Just spent hours trying to figure out what was wrong because the rc is released via the latest tag. Please fix before releasing v7 and please put v6 on the latest tag.

Gotta agree here. v7 really shouldn't be marked as the latest version on npm.

**Comment by daniloab** [comment]
the npm latest version was fixed for the latest one: 6.1.0 as @lachieh said.

We advise swagger-jsdoc users to use it since is stabilized and running perfectly.

You can check here https://www.npmjs.com/package/swagger-jsdoc

**Comment by daniloab** [comment]
@a-dams154 bug from #270 using the release candidate version, it is very similar. Let's add here to have a robust track about it:

`error code : 'ERR_REQUIRE_ESM'`
![image](https://user-images.githubusercontent.com/19939822/120325584-2ed1ac80-c2be-11eb-9be0-8162cf5298dc.png)

**Comment by flutterq2** [comment]
[[Already Solved] Error [ERR_REQUIRE_ESM]: Must use import to load ES Module
](https://exerror.com/error-err_require_esm-must-use-import-to-load-es-module/)
